### PR TITLE
Add validation_split to fit_generator #3900

### PR DESCRIPTION
**Pull request by kouml**
This is an alternative solution for adding validation_split to fit_generator(#3900 ).
I made utility which can split train and validation from original_dir with validation_split argument.

I'd like to get your feedback.
If it is not fit keras API, please close.

```python
train_dir, val_dir = keras.utils.data_utils.train_valid_split(original_dir, validation_split=0.1)
# all data in train_dir which are alias to original_data.
# and train_dir is a temporary directory.

train_datagen = ImageDataGenerator(
        rescale=1./255,
        shear_range=0.2,
        zoom_range=0.2,
        horizontal_flip=True)

val_datagen = ImageDataGenerator(rescale=1./255)

train_generator = train_datagen.flow_from_directory(
        train_dir,
        target_size=(150, 150),
        batch_size=32,
        class_mode='binary')

val_generator = val_datagen.flow_from_directory(
       val_dir,
        target_size=(150, 150),
        batch_size=32,
        class_mode='binary')
```

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. This API looks fine, but since we have already merged this feature in the API, we will not pursue this alternative.

In terms of implementation, the use of `os.symlink` on a lot of files seems risky (and the user may not always have the rights to create them, since it is relatively common to read files from a shared read-only directory). The symlinks are also seemingly never destroyed?

**Comment by kouml** [comment]
Thanks for the review. 
I think this implementation has a benefit that can customize for each generator.
In my opinion, it can be easily handling depending on whether or not you keep tempfile object.
however, since other solution is already merged, this PR should close.

is usage like this?
```
m = MobileNet()
datagen = ImageDataGenerator(validation_split=0.2)
t = datagen.flow_from_directory(
    'cifar',
    target_size=(224, 224, 3),
    subset='training'
    )

v = datagen.flow_from_directory(
    'cifar',
    target_size=(224, 224, 3),
    subset='validation'
    )
```

**Comment by fchollet** [comment]
Yes, that's correct.

Closing the PR then.

**Comment by lAlexMel** [comment]
v = datagen.flow_from_directory(
    'cifar',
    target_size=(224, 224, 3),
    subset='validation'
    )

Now, I want to look at the fifth image into validation set. How I can?
